manifests/daemon.pp

```diff
@@ -82,6 +82,7 @@
   String $service_ensure               = 'running',
   Boolean $service_enable              = true,
   Boolean $manage_service              = true,
+  Boolean $manage_systemd_unit         = $install_method != 'package',
```

Comment:
I never know if this actually works or if it's trying to be too clever. Usually I play it safe, set it to `undef` and use `pick($manage_systemd_unit, $install_method != 'package')`

Reply:
not being familiar with the `pick` function, I don't actually know how you would expect me to use this. :)

Reply:
https://github.com/puppetlabs/puppetlabs-stdlib/blob/master/REFERENCE.md#pick

My biggest concern is that your implementation relies on a certain compile order and I don't know if Puppet actually guarantees that it works. If you override install_method via hiera, does it also still work?